pkg/bundle/bundle.go

```diff
@@ -202,7 +202,26 @@ func (b *bundle) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result,
 			return ctrl.Result{Requeue: true}, b.targetDirectClient.Status().Update(ctx, &bundle)
 		}
 
-		if synced {
+        secretSynced := false
+        if bundle.Spec.Target.Secret != nil {
```

Comment:
suggestion: Since Secrets and ConfigMaps as targets would be mutually exclusive, we should probably avoid doing the sync with the ConfigMap above if we have a Secret target set. i.e., above we'd add something like `if bundle.Spec.Target.ConfigMap != nil`

Reply:
There is a misunderstanding on the intent.
I did not intend to have `target.secret` and `target.configmap` mutually exclusive.
In fact, in my usecase, I want both targets populated by the same bundle.

If you feel this should be mutually exclusive to simplify error handling, then I can do the necessary changes.

Reply:
Even if we don't make them mutually exclusive, we'll still need to change the validation because it requires that a configmap is set. That would make it impossible to have a Bundle which only targets a secret.

See this line: https://github.com/cert-manager/trust-manager/blob/60150c86bd873c5fa02752b63723920e4bf70b72/pkg/webhook/validation.go#L168-L169